File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        // 1.
        PriorityQueue<Integer> queue = new PriorityQueue<>(Collections.reverseOrder());
        for(int p: priorities){
            queue.add(p);
        }
        
        // 2.
        int answer = 0;
        while(!queue.isEmpty()){
            for(int i=0; i<priorities.length; i++){
                if(priorities[i] == queue.peek()){
                    queue.poll();
                    answer++;
                    if(i == location){
                        return answer;
                    }
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    v = max(priorities)
    q = deque([])
    for i in range(len(priorities)):
        q.append((priorities[i], i))
     
    answer = 0
    while q:
        temp = q.popleft()
        
        if temp[0] < v:
            q.append(temp)
            
        elif temp[0] == v:            
            answer += 1
            if temp[1] == location:
                return answer
            
            if len(priorities) > 1:
                priorities.remove(v)
                v = max(priorities)